sycl/plugins/unified_runtime/ur/adapters/cuda/context.cpp

```diff
@@ -0,0 +1,154 @@
+//===--------- context.cpp - CUDA Adapter ----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+
+#include "context.hpp"
+
+#include <cassert>
+
+/// Create a UR CUDA context.
+///
+/// By default creates a scoped context and keeps the last active CUDA context
+/// on top of the CUDA context stack.
+/// With the __SYCL_PI_CONTEXT_PROPERTIES_CUDA_PRIMARY key/id and a value of
+/// PI_TRUE creates a primary CUDA context and activates it on the CUDA context
+/// stack.
+///
+UR_APIEXPORT ur_result_t UR_APICALL
+urContextCreate(uint32_t DeviceCount, const ur_device_handle_t *phDevices,
+                const ur_context_properties_t *pProperties,
+                ur_context_handle_t *phContext) {
+  std::ignore = DeviceCount;
+  std::ignore = pProperties;
+  UR_ASSERT(phDevices, UR_RESULT_ERROR_INVALID_NULL_POINTER);
```

Comment:
FYI: we are not doing any of these checks in the UR L0 adapter, since those are done by the validation layer in the UR loader. only issue with doing them here as well is extra instructions being executed (minimal performance hit, but stil).

Reply:
That makes sense, we've been running the UR conformance tests without the validation layer so needed these checks to pass the tests. We can address this in a follow-up PR.